Ah, there is a build failure on windows:
```
C:\projects\solidity\test\yulPhaser\Population.cpp(147): error C3493: 'maxLength' cannot be implicitly captured because no default capture mode has been specified [C:\projects\solidity\build\test\soltest.vcxproj]
C:\projects\solidity\test\yulPhaser\Population.cpp(148): error C2665: 'solidity::phaser::Population::makeRandom': none of the 2 overloads could convert all the argument types [C:\projects\solidity\build\test\soltest.vcxproj]
  C:\projects\solidity\tools/yulPhaser/Population.h(78): note: could be 'solidity::phaser::Population solidity::phaser::Population::makeRandom(solidity::phaser::Program,size_t,std::function<size_t (void)>)'
  C:\projects\solidity\test\yulPhaser\Population.cpp(148): note: while trying to match the argument list '(solidity::phaser::Program, const size_t, int)'
C:\projects\solidity\test\yulPhaser\Population.cpp(159): error C3536: 'population': cannot be used before it is initialized [C:\projects\solidity\build\test\soltest.vcxproj]
C:\projects\solidity\test\yulPhaser\Population.cpp(153): error C2672: 'count_if': no matching overloaded function found [C:\projects\solidity\build\test\soltest.vcxproj]
C:\projects\solidity\test\yulPhaser\Population.cpp(159): error C2780: 'iterator_traits<_FwdIt>::difference_type std::count_if(_ExPo &&,const _FwdIt,const _FwdIt,_Pr) noexcept': expects 4 arguments - 2 provided [C:\projects\solidity\build\test\soltest.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\algorithm(291): note: see declaration of 'std::count_if'
C:\projects\solidity\test\yulPhaser\Population.cpp(159): error C2780: 'iterator_traits<_Iter>::difference_type std::count_if(_InIt,_InIt,_Pr)': expects 3 arguments - 2 provided [C:\projects\solidity\build\test\soltest.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\algorithm(269): note: see declaration of 'std::count_if'
```

--- a/test/yulPhaser/Common.cpp
+++ b/test/yulPhaser/Common.cpp
@@ -25,6 +25,15 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
 
+vector<size_t> phaser::test::chromosomeLengths(Population const& _population)
+{
+	vector<size_t> lengths;
+	for (auto const& individual: _population.individuals())
+		lengths.push_back(individual.chromosome.length());
+
+	return lengths;
+}
+
 map<string, size_t> phaser::test::enumerateOptmisationSteps()
 {
 	map<string, size_t> stepIndices;

--- a/test/yulPhaser/Common.cpp
+++ b/test/yulPhaser/Common.cpp
@@ -19,6 +19,8 @@
 
 #include <libyul/optimiser/Suite.h>
 
+#include <regex>
+
 using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
@@ -31,4 +33,10 @@ map<string, size_t> phaser::test::enumerateOptmisationSteps()
 		stepIndices.insert({nameAndAbbreviation.first, i++});
 
 	return stepIndices;
+}
+
+string phaser::test::stripWhitespace(string const& input)
+{
+	regex whitespaceRegex("\\s+");
+	return regex_replace(input, whitespaceRegex, "");
 }

--- a/test/yulPhaser/Common.h
+++ b/test/yulPhaser/Common.h
@@ -28,6 +28,8 @@
 
 #pragma once
 
+#include <tools/yulPhaser/Population.h>
+
 #include <cassert>
 #include <map>
 #include <string>
@@ -37,6 +39,10 @@ namespace solidity::phaser::test
 {
 
 // CHROMOSOME AND POPULATION HELPERS
+
+/// Returns a vector containing lengths of all chromosomes in the population (in the same order).
+std::vector<size_t> chromosomeLengths(Population const& _population);
+
 /// Assigns indices from 0 to N to all optimisation steps available in the OptimiserSuite.
 /// This is a convenience helper to make it easier to test their distribution with tools made for
 /// integers.

--- a/test/yulPhaser/Common.h
+++ b/test/yulPhaser/Common.h
@@ -42,6 +42,11 @@ namespace solidity::phaser::test
 /// integers.
 std::map<std::string, size_t> enumerateOptmisationSteps();
 
+// STRING UTILITIES
+
+/// Returns the input string with all the whitespace characters (spaces, line endings, etc.) removed.
+std::string stripWhitespace(std::string const& input);
+
 // STATISTICAL UTILITIES
 
 /// Calculates the mean value of a series of samples given in a vector.

--- a/test/yulPhaser/CommonTest.cpp
+++ b/test/yulPhaser/CommonTest.cpp
@@ -51,6 +51,20 @@ BOOST_AUTO_TEST_CASE(enumerateOptimisationSteps_should_assing_indices_to_all_ava
 	}
 }
 
+BOOST_AUTO_TEST_CASE(stripWhitespace_should_remove_all_whitespace_characters_from_a_string)
+{
+	BOOST_TEST(stripWhitespace("") == "");
+	BOOST_TEST(stripWhitespace(" ") == "");
+	BOOST_TEST(stripWhitespace(" \n\t\v ") == "");
+
+	BOOST_TEST(stripWhitespace("abc") == "abc");
+	BOOST_TEST(stripWhitespace(" abc") == "abc");
+	BOOST_TEST(stripWhitespace("abc ") == "abc");
+	BOOST_TEST(stripWhitespace(" a b c ") == "abc");
+	BOOST_TEST(stripWhitespace(" a b\tc\n") == "abc");
+	BOOST_TEST(stripWhitespace("   a   b \n\n  c \n\t\v") == "abc");
+}
+
 BOOST_AUTO_TEST_CASE(mean_should_calculate_statistical_mean)
 {
 	BOOST_TEST(mean<int>({0}) == 0.0);

--- a/test/yulPhaser/CommonTest.cpp
+++ b/test/yulPhaser/CommonTest.cpp
@@ -19,11 +19,14 @@
 
 #include <libyul/optimiser/Suite.h>
 
+#include <liblangutil/CharStream.h>
+
 #include <boost/test/unit_test.hpp>
 
 #include <set>
 
 using namespace std;
+using namespace solidity::langutil;
 using namespace solidity::yul;
 using namespace boost::test_tools;
 
@@ -32,6 +35,18 @@ namespace solidity::phaser::test
 
 BOOST_AUTO_TEST_SUITE(Phaser)
 BOOST_AUTO_TEST_SUITE(CommonTest)
+
+BOOST_AUTO_TEST_CASE(chromosomeLengths_should_return_lengths_of_all_chromosomes_in_a_population)
+{
+	CharStream sourceStream("{}", "");
+	auto program = Program::load(sourceStream);
+
+	Population population1(program, {Chromosome(), Chromosome("a"), Chromosome("aa"), Chromosome("aaa")});
+	BOOST_TEST((chromosomeLengths(population1) == vector<size_t>{0, 1, 2, 3}));
+
+	Population population2(program);
+	BOOST_TEST((chromosomeLengths(population2) == vector<size_t>{}));
+}
 
 BOOST_AUTO_TEST_CASE(enumerateOptimisationSteps_should_assing_indices_to_all_available_optimisation_steps)
 {

--- a/test/yulPhaser/Program.cpp
+++ b/test/yulPhaser/Program.cpp
@@ -15,6 +15,8 @@
 	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <test/yulPhaser/Common.h>
+
 #include <tools/yulPhaser/Exceptions.h>
 #include <tools/yulPhaser/Program.h>
 
@@ -30,7 +32,6 @@
 #include <boost/test/unit_test.hpp>
 
 #include <cassert>
-#include <regex>
 #include <string>
 
 using namespace std;
@@ -50,12 +51,6 @@ namespace
 			return skipRedundantBlocks(get<Block>(_block.statements[0]));
 		else
 			return _block;
-	}
-
-	string stripWhitespace(string const& input)
-	{
-		regex whitespaceRegex("\\s+");
-		return regex_replace(input, whitespaceRegex, "");
 	}
 }
 

--- a/tools/yulPhaser/Population.h
+++ b/tools/yulPhaser/Population.h
@@ -67,7 +67,7 @@ public:
 	friend std::ostream& operator<<(std::ostream& _stream, Population const& _population);
 
 private:
-	explicit Population(Program _program, std::vector<Individual> _individuals = {}):
+	explicit Population(Program _program, std::vector<Individual> _individuals):
 		m_program{std::move(_program)},
 		m_individuals{std::move(_individuals)} {}
 

